tests/pytests/functional/modules/test_win_file.py

```diff
@@ -0,0 +1,439 @@
+import ctypes
+import os
+import platform
+
+import pytest
+
+import salt.modules.win_useradd
+import salt.utils.files
+from salt.exceptions import CommandExecutionError
+from salt.modules import win_file
+
+pytestmark = [
+    pytest.mark.windows_whitelisted,
+    pytest.mark.skip_unless_on_windows,
+]
+
+
+@pytest.fixture
+def configure_loader_modules(minion_opts, modules):
+    utils = salt.loader.utils(minion_opts)
+    return {
+        win_file: {
+            "__salt__": modules,
+            "__utils__": utils,
+        },
+    }
+
+
+skip_not_windows_admin = pytest.mark.skipif(
+    platform.system() != "Windows" or not ctypes.windll.shell32.IsUserAnAdmin(),
+    reason="user needs windows admin rights",
+)
+
+
+@pytest.fixture
+def windows_user():
+    return salt.modules.win_useradd.current()
+
+
+@skip_not_windows_admin
+def test_symlink(tmp_path):
+    tmp_path = str(tmp_path)
+    file = os.path.join(tmp_path, "t.txt")
+    with salt.utils.files.fopen(file, "w"):
+        pass
+    assert os.path.isfile(file) is True
+    link = os.path.join(tmp_path, "l")
+    assert win_file.is_link(link) is False
+    assert win_file.symlink(file, link) is True
+    assert os.path.isfile(file) is True
+    assert win_file.is_link(link) is True
```

Comment:
```suggestion
    file = tmp_path.joinpath("t.txt").touch()
    assert file.is_file() is True
    link = tmp_path / "l"
    assert win_file.is_link(link) is False
    assert win_file.symlink(file, link) is True
    assert file.is_file() is True
    assert win_file.is_link(link) is True
```
Same for remaining tests.